### PR TITLE
Update copyrights

### PR DESCRIPTION
**Pull request by peabody124**
Include dRonin in copyrights in header where appropriate. May be outdated by
#2155 but best to do the right thing for now.

To be added: 
- [x] https://github.com/TauLabs/TauLabs/pull/2175
- [x] https://github.com/TauLabs/TauLabs/pull/2178
- [x] https://github.com/TauLabs/TauLabs/pull/2179
- [x] https://github.com/TauLabs/TauLabs/pull/2181

### CODE REVIEW

**Comment by mlyle** [comment]
Thank you.  Hopefully we can stay on a friendlier footing for now on.

**Comment by tracernz** [comment]
Thanks @peabody124, much appreciated. Would be nice to have these discussions on IRC so we can be clear on each others intentions and reasoning in future, before things get out of hand. :+1:

**Comment by kubark42** [comment]
@tracernz Seeing how emotional this was for dRonin, I don't think that would have helped. We did what we said we would, in an appropriate time frame. Why the drama, I'll never understand.

@peabody124 Thanks. When I get some bandwidth I'll make sure we caught them all and merge this.

**Comment by mlyle** [comment]
@kubark42 --- I asked / raised this issue well before the first PR was open; we raised it in the initial PR; we asked before the first merge, and we brought it up calmly.  Then you guys tried to come up with a way to avoid attributing at all ;)

An appropriate timeframe is "at the same time as the merge".  The tree should not be left without the notices.

File: flight/targets/sparky/fw/pios_board.c [hunk 1]
```diff
@@ -6,7 +6,8 @@
  * @{
  *
  * @file       pios_board.c
- * @author     Tau Labs, http://taulabs.org, Copyright (C) 2012-2014
+ * @author     dRonin, http://dronin.org Copyright (C) 2015
```
Comment (kubark42) [inline flight/targets/sparky/fw/pios_board.c:9]: https://github.com/peabody124/TauLabs/commits/copyrights/flight/targets/sparky/fw/pios_board.c shows that this doesn't have any recent changes. Included by accident?

**Comment by kubark42** [comment]
@mlyle Feel free to point to me where we tried to avoid attributing anything. The best analysis is when we didn't follow your demands hysterics ensued. Anyone is free to read GitHub discussions and judge for themselves.

@peabody124 I dunno about this PR. It really looks as if many of the changesets merged in are trivial in nature, and so don't warrant an attribution. Indeed, in addition to the spiritual violations you mentioned in another thread, it looks as if dRonin is claiming copyright of files which have had no qualifying changes. See https://github.com/d-ronin/dRonin/commits/698a1b406e6aa9cd8e91e593d0925048b7501076/flight/targets/sparky2/board-info/board_hw_defs.c, where the only dRonin change was of a non-creative nature. 

Frankly, I really don't care about dRonin falsely claiming authorship. It's not worth getting worked up over, and the git commit logs are all that really matters. So I'm proposing we set a good example and not get on dRonin's case about their attribution violations. I also propose to merge this PR without going through and rejecting authorship when only trivial changes were made. This seems to respect the open-source spirit of ensuring that attribution is spread as widely as possible, even though legally it's not required.

File: flight/targets/sparky/fw/pios_board.c (hunk 1 above)
Comment (mlyle) [inline flight/targets/sparky/fw/pios_board.c:9]: https://github.com/TauLabs/TauLabs/pull/2145/files

**Comment by mlyle** [comment]
>  looks as if dRonin is claiming copyright of files which have had no qualifying changes. 

Whether this is "qualifying" or not is the kind of thing that different US federal circuit courts have decided differently.  In any case, there's no harm in adding the notice-- but potential risks to anyone who decides to remove it. :)

**Comment by peabody124** [comment]
@kubark42 I'd rather err on the side of caution and just do more rather than less. When we proposed going to a universal file we were basically going to give equal attribution at the top level. As you pointed out the git commit history is the only meaningful thing to see fraction of contribution. When there is a DR header in there and 99.9% of the LOC in the codebase are from TL and OP, well it is what it is :)

**Comment by mlyle** [comment]
>  When there is a DR header in there and 99.9% of the LOC in the codebase are from TL and OP, well it is what it is :)

Yah, it may end up that way because I assume it's going to get more difficult to take our changesets with time.  But if you continue tracking that close, it won't; right now about 15% of flight blames to dRonin in our tree.

**Comment by kubark42** [comment]
<Shrug> You're sloppy in your attribution. Consider https://github.com/d-ronin/dRonin/pull/569/files#diff-12bc03d72ddbc63727724364b733ca3a, where the only change was to rename "Throttle" to the almost synonym "Thrust". It's pretty clear that when you move files around, rename variables, add new hardware targets (which are 99% boilerplate copy pasted from original targets), and liberally sprinkle in authorship lines you'll quickly bulk up a figure. It's just hard to believe based on the PRs I've seen that in two months you've replaced 15% of the functionality built in over the past 6 years. 

These arguments thrown at us are so slippery. First it was absolute legality, then when that bubble was burst it was morality, and then now that that bubble is burst it's `git blame`. It was clear from the beginning that this discussion was unnecessary, and nothing has come out of it except a better understanding of why projects should use centralized copyright. (Shout out to @peabody124 for discovering that!)

As always, feel free to point out to us when you'd like an attribution to be added and we'll ensure it does, even when it's not legally appropriate. That just seems like the right thing to do.

**Comment by mlyle** [comment]
>  You're sloppy in your attribution. 

If we want, we can add a Copyright dRonin to every file-- and perhaps we should, to avoid this type of discussion.  It might be a good idea.  The purpose of the copyright notice isn't just project-level attribution.  It's also to let people know that there (may) be a copyright interest held in that file by the project.

> As always, feel free to point out to us when you'd like an attribution to be added and we'll ensure it does, even when it's not legally appropriate. That just seems like the right thing to do.

As always, when you import from our project, ensure that you preserve the notices, as you are legally required (and now as we have very explicitly clarified.)  If you slip up a time or two, no big deal.  If you systematically ignore it (as you still are, by still not merging this PR), ... I will have to consider all remedies afforded to me by the license agreements and copyright law.

**Comment by mlyle** [comment]
I mean, you say it's the right thing to do?  Go ahead, press merge, and include it directly in subsequent/new PRs.

**Comment by kubark42** [comment]
The right thing to do is stick with the original plan. I'm unsure why you consciously refuse to hear this, but here it is again: once I'm done reviewing #2127/#2134 I'll merge this all in in one fell swoop. 

We're making steady progress. Hold tight.

**Comment by mlyle** [comment]
In the meantime you're still distributing stuff without the copyright notices attached.

**Pull request merged**